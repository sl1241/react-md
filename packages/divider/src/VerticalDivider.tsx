import { $$ } from "voby"
import { Divider } from "./Divider"
import { useVerticalDividerHeight } from "./useVerticalDividerHeight"

export interface VerticalDividerProps extends HTMLAttributes<HTMLDivElement> {
    /** {@inheritDoc VerticalDividerHookOptions.maxHeight} */
    maxHeight?: FunctionMaybe<Nullable<number>>
}

/**
 * This component is used to create a vertical divider based on a parent
 * element's height. This is really only needed when the parent element **has no
 * defined height**.  If there is a defined height, this component is not worth
 * much as the height can be computed in css as normal. This really just fixes
 * the issue that the height would be set to `auto` (which computes to 0 most of
 * the time) when it is not set on a parent element.
 */
export const VerticalDivider = ({ style, maxHeight = 1, ref, ...props }: VerticalDividerProps) => {
    const heightProps = useVerticalDividerHeight({ ref, style, maxHeight: $$(maxHeight), })

    //@ts-ignore
    return <Divider {...props} {...heightProps} vertical />
}
