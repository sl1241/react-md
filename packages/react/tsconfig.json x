{
  "extends": "../../tsconfig.base.json",
  "compilerOptions": {
    "outDir": "./dist",
    "rootDir": "./src",
    "declarationDir": "./dist/types",
    "composite": true,
    "skipLibCheck": true,
    "skipDefaultLibCheck": true,
    "jsx": "preserve",
    "jsxImportSource": "voby",
    "types": [
      "voby/dist/jsx/jsx"
    ],
  },
  "exclude": [
    "vite.config.ts",
"vite.config.umd.ts",
    "types",
  ],
}
