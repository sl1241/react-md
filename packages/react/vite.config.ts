import { defineConfig } from 'vite'

const config = defineConfig({
    build: {
        minify: false,
        lib: {
            entry: ["./src/index.ts"],
            name: "react",
            formats: ['cjs', 'es', 'umd'],
            fileName: (format: string, entryName: string) => `${entryName}.${format}.js`
        },
        // rollupOptions: {
        //   output: {
        //     preserveModules: true,
        //   }
        // },
    },


    esbuild: {
        jsx: 'automatic',
    },

    css: {
        modules: {
            scopeBehaviour: 'global',
            localsConvention: 'camelCase',
        }
    },
})



export default config
