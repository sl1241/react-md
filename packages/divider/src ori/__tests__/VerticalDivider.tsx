import { render } from "@testing-library/react";

import { VerticalDivider } from "../VerticalDivider";

describe("VerticalDivider", () => {
  it("should render as a div with the vertical divider class names", () => {
    const { container } = render(<VerticalDivider />);
    const divider = container.querySelector(".rmd-divider") as HTMLDivElement;

    expect(divider.tagName).toBe("DIV");
    expect(divider.className).toBe("rmd-divider rmd-divider--vertical");
    expect(container).toMatchSnapshot();
  });
});
