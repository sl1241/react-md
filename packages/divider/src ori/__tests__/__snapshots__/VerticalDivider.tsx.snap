// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VerticalDivider should render as a div with the vertical divider class names 1`] = `
<div>
  <div
    class="rmd-divider rmd-divider--vertical"
    role="separator"
    style="height: 0px;"
  />
</div>
`;
